crates/ra_hir_ty/src/tests/traits.rs

```diff
@@ -3029,3 +3029,34 @@ fn infer_dyn_fn_output() {
     "###
     );
 }
+
+#[test]
+fn variable_kinds() {
+    assert_snapshot!(
+        infer(
+            r#"
+trait Trait<T> { fn get(self, t: T) -> T; }
+struct S;
+
+impl Trait<u128> for S {}
+impl Trait<f32> for S {}
+
+fn test() {
+    S.get(1);
+    S.get(1.);
+}
+        "#
+        ),
+        @r###"
+    24..28 'self': Self
+    30..31 't': T
+    117..149 '{     ...1.); }': ()
+    123..124 'S': S
+    123..131 'S.get(1)': u128
+    129..130 '1': u128
+    137..138 'S': S
+    137..146 'S.get(1.)': f32
+    143..145 '1.': f32
+    "###
+    );
```

Comment:
```suggestion

        check_types(
            r#"
trait Trait<T> { fn get(self, t: T) -> T; }
struct S;

impl Trait<u128> for S {}
impl Trait<f32> for S {}

fn test() {
    S.get(1);
  //^^^^^^^^ u128
    S.get(1.);
  //^^^^^^^^ f32
}
        "#
        );
);
```

Reply:
Ah, the original also asserts that literals are assigned the right type. You can (but I am not sure if that's reasonable) to do 

```
 S.get(  1  );
//^u128  ^u128 
```